src/mumble/GlobalShortcut_win.cpp

```diff
@@ -719,211 +421,170 @@ void GlobalShortcutWin::timeTicked() {
 				// However, since the buttons field is only a 16-bit value,
 				// and we also want to use the left and right triggers as
 				// buttons, we assign them the button indexes 16 and 17.
-				uint32_t buttonMask = state.buttons;
-				for (uint32_t j = 0; j < 18; j++) {
-					QList< QVariant > ql;
-
+				for (uint8_t j = 0; j < 18; ++j) {
 					bool pressed = false;
-					if (j >= 16) {
-						if (j == 16) { // LeftTrigger
+					switch (j) {
+						case 16:
 							pressed = state.leftTrigger > XBOXINPUT_TRIGGER_THRESHOLD;
-						} else if (j == 17) { // RightTrigger
+							break;
+						case 17:
 							pressed = state.rightTrigger > XBOXINPUT_TRIGGER_THRESHOLD;
-						}
-					} else {
-						uint32_t currentButtonMask = (1 << j);
-						pressed                    = (buttonMask & currentButtonMask) != 0;
+							break;
+						default:
+							pressed = (state.buttons & (1 << j)) != 0;
 					}
 
-					uint32_t type = (i << 24) | j;
-					ql << static_cast< uint >(type);
-					ql << XboxInput::s_XboxInputGuid;
+					const QList< QVariant > ql{ QVariant::fromValue(DeviceTypes::XINPUT), (i << 24) | j };
```

Comment:
What is this bit magic doing here? :eyes:

Reply:
https://github.com/mumble-voip/mumble/blob/822185e99794d8606a1c52049f66fd9678b995e7/src/mumble/GlobalShortcut_win.cpp#L706-L721

Reply:
For the sake of readability I would factor this bit-magic with a description (or a reference to where to find such description) out into its own function with an expressive name.

Reply:
The comment is right above the relevant block, though.

Reply:
Factoring this out into a function will still improve readability

Reply:
I agree.